gt/communication/ghex_comm_mt.hpp

```diff
@@ -0,0 +1,179 @@
+#pragma once
+
+#include "./ghex_comm.hpp"
+#include "./factorize.hpp"
+#include <ghex/glue/gridtools/field.hpp>
+
+namespace communication {
+
+namespace ghex_comm {
+
+
+class grid_mt
+{
+public: // member types
+  using domain_id_type    = local_domain::domain_id_type;
+  using transport_type    = ::gridtools::ghex::tl::mpi_tag;
+  using grid_type         = ::gridtools::ghex::structured::grid::template type<local_domain>;
+  using patterns_type     = ::gridtools::ghex::pattern_container<transport_type, grid_type, domain_id_type>;
+  using patterns_ptr_t    = std::unique_ptr<patterns_type>;
+
+protected: // member types
+    using coordinate_type = local_domain::coordinate_type;
+    using domain_vec      = std::vector<local_domain>;
+    using comm_obj_type   = typename std::remove_reference<typename std::remove_cv<decltype(
```

Comment:
we are using c++14, so you can use `std::remove_reference_t` and friends

Reply:
ok